presto-parser/src/main/java/com/facebook/presto/sql/tree/RoutineCharacteristics.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.tree;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.facebook.presto.sql.tree.RoutineCharacteristics.Determinism.DETERMINISTIC;
+import static com.facebook.presto.sql.tree.RoutineCharacteristics.Determinism.NOT_DETERMINISTIC;
+import static com.facebook.presto.sql.tree.RoutineCharacteristics.Language.SQL;
+import static com.facebook.presto.sql.tree.RoutineCharacteristics.NullCallClause.CALLED_ON_NULL_INPUT;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class RoutineCharacteristics
+{
+    public enum Language
+    {
+        SQL;
+    }
+
+    public enum Determinism
+    {
+        DETERMINISTIC,
+        NOT_DETERMINISTIC;
+    }
+
+    public enum NullCallClause
+    {
+        RETURNS_NULL_ON_NULL_INPUT,
+        CALLED_ON_NULL_INPUT;
+    }
+
+    private final Language language;
+    private final Determinism determinism;
+    private final NullCallClause nullCallClause;
+
+    public RoutineCharacteristics(
```

Comment:
This constructor is unused. Also the semantics are not obvious. Let's remove it.

Reply:
This is called in `AstBuilder` I think. It's used to provide default values when they are missing.